### PR TITLE
Init Azure.ResourceManager.Reservations

### LINKED ISSUE #28687: Initial Azure.ResourceManager.Reservations

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/Price.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.Reservations.Models
+{
+    /// <summary> The Price. </summary>
+    public partial class Price
```
Comment (Yao725) [inline sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/Price.cs:11]: Can we change this single class name?

File: sdk/reservations/Azure.ResourceManager.Reservations/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Azure reservations Management client library for .NET
```
Comment (Yao725) [inline sdk/reservations/Azure.ResourceManager.Reservations/README.md:1]: `Reservations`?

File: sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/ResourceType.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Reservations.Models
+{
+    /// <summary> The resource types. </summary>
+    public readonly partial struct ResourceType : IEquatable<ResourceType>
```
Comment (Yao725) [inline sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/ResourceType.cs:14]: I think we'd better avoid using `ResourceType` as we already has a definition in `Azure.Core`.

File: sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/Price.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/Price.cs]: rename to PurchasePrice

File: sdk/reservations/Azure.ResourceManager.Reservations/README.md (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/reservations/Azure.ResourceManager.Reservations/README.md]: Fixed

File: sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/ResourceType.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/Models/ResourceType.cs]: rename to ResourceTypeName

**Review by Yao725 — APPROVED** [review]

**Pull request merged**